rules/prefer-type-error.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
```

Comment:
Checking if something is finite doesn't imply that the type is wrong imo. What do you guys think?